src/Goteo/Controller/Api/DataSetApiController.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/*
+ * This file is part of the Goteo Package.
+ *
+ * (c) Platoniq y Fundación Goteo <fundacion@goteo.org>
+ *
+ * For the full copyright and license information, please view the README.md
+ * and LICENSE files that was distributed with this source code.
+ */
+
+namespace Goteo\Controller\Api;
+
+use Goteo\Application\View;
+use Goteo\Entity\DataSet;
+use Goteo\Repository\DataSetRepository;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+class DataSetApiController extends AbstractApiController
+{
+    public function dataSetsAction(Request $request): JsonResponse {
+        $filters = [];
+
+        $page = $request->query->has('page') ?? 0;
+        $limit = $request->query->get('limit') ?? 10;
```

Comment:
@davidbeig just FYI, the `Request` class has a simpler mechanism that allows to get a query parameter with a default value if it fails to get it.

So, this could be written as well as:

```
$page = $request->get('page', 0);
$limit = $request->get('limit', 10);
```

Reply:
True!

https://github.com/symfony/symfony/blob/0a9296a8b23360475c282cd98b1fd36e3937d259/src/Symfony/Component/HttpFoundation/Request.php#L819

I have just looked at it, I didn't know!